src/frontend/src/session.rs

```diff
@@ -458,15 +468,20 @@ pub struct SessionImpl {
     /// Identified by process_id, secret_key. Corresponds to SessionManager.
     id: (i32, i32),
 
+    /// Client address
+    peer_addr: SocketAddr,
+
     /// Transaction state.
-    // TODO: get rid of the `Mutex` here as a workaround if the `Send` requirement of
-    // async functions, there should actually be no contention.
+    /// TODO: get rid of the `Mutex` here as a workaround if the `Send` requirement of
+    /// async functions, there should actually be no contention.
     txn: Arc<Mutex<transaction::State>>,
 
     /// Query cancel flag.
     /// This flag is set only when current query is executed in local mode, and used to cancel
     /// local query.
     current_query_cancel_flag: Mutex<Option<ShutdownSender>>,
+
+    exec_context: Mutex<Option<Weak<ExecContext>>>,
```

Comment:
Why using a `Weak` reference here? (Is a `Session` possible to be leaked?)

Reply:
Each session could run different SQLs multiple times. In my mind `exec_context ` represents the lifetime of a running SQL in the current session. Once the `Arc< ExecContext >` is dropped, we consider the running SQL ends. cc @BugenZhao how do you think?

Reply:
My original idea is to leverage `Drop`, like this:
https://github.com/risingwavelabs/risingwave/blob/ed4a3d7bb41f861ce11737c46018a4c4cf1bec5c/src/frontend/src/session/transaction.rs#L90-L104

However, it seems also okay to let `Arc` handle that for us. We may make the inner field private to avoid leaking the ref count.